src/Repository/WorklogRepository.php

```diff
@@ -133,4 +134,42 @@ public function findBillableWorklogsByWorkerAndDateRange(string $workerIdentifie
             ])
             ->getQuery()->getResult();
     }
+
+    /**
+     * @throws \Exception
+     */
+    public function getWorklogsAttachedToInvoiceInDateRange(\DateTimeInterface $periodStart, \DateTimeInterface $periodEnd, int $page = 1, int $pageSize = 50): array
+    {
+        $from = new \DateTimeImmutable($periodStart->format('Y-m-d').' 00:00:00');
+        $to = new \DateTimeImmutable($periodEnd->format('Y-m-d').' 23:59:59');
+
+        $query = $this->createQueryBuilder('worklog')
+            ->leftJoin(Issue::class, 'issue', 'WITH', 'worklog.issue = issue.id')
+            ->leftJoin(Project::class, 'project', 'WITH', 'issue.project = project.id')
+            ->where('worklog.invoiceEntry IS NOT NULL')
+            ->andWhere('worklog.started BETWEEN :from AND :to')
+            ->setParameter('from', $from)
+            ->setParameter('to', $to)
+            ->getQuery()
+            ->setFirstResult(($page - 1) * $pageSize)
+            ->setMaxResults($pageSize);
+
+        $paginator = new Paginator($query, true);
+
+        $totalItemCount = count($paginator);
+        $pagesCount = ceil($totalItemCount / $pageSize);
+
+        $items = [];
```

Comment:
why not just return the paginator instead of this conversion to array? This results in unnecessary iterations.

```
return [
    'total_count' => $totalItemCount,
    'pages_count' => $pagesCount,
    'current_page' => $page,
    'page_size' => $pageSize,
    'paginator' => $paginator,
];
```

you can iterate it the same way as an array

`foreach ($invoiceAttachedWorklogs['paginator'] as $worklog) {`